GREEDY_WEB/ClientApp/components/UserRegistration.tsx

```diff
@@ -0,0 +1,203 @@
+﻿import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { Button } from 'reactstrap';
+import DocumentTitle from 'react-document-title';
+import { RegistrationForm } from 'react-stormpath';
+import axios from 'axios';
+import { Link, NavLink } from 'react-router-dom';
+import Cookies from 'universal-cookie';
+import FacebookLogin from 'react-facebook-login';
+
+interface State {
+    isAccountCreated: any;
+}
+
+export class UserRegistration extends React.Component<RouteComponentProps<{}>, State> {
+    constructor(props) {
+        super(props);
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+        this.state = {
+            isAccountCreated: false
+        }
+    }
+
+    onFormSubmit(e, next) {
+        e.preventDefault();
+        var data = e.data;
+
+        if (data.username.length < 1) {
+            return next(new Error('Username must not be empty.'));
+        }
+
+        if (data.email.length < 1) {
+            return next(new Error('Email must not be empty.'));
+        }
+
+        var regex = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/;
+        if (!regex.test(data.email)) {
+            return next(new Error('Wrong email.'));
+        }
+
+        if (data.givenName.length < 1) {
+            return next(new Error('First name must not be empty.'));
+        }
+
+        if (data.surname.length < 1) {
+            return next(new Error('Last name must not be empty.'));
+        }
+
+        if (data.givenName.length > 256) {
+            return next(new Error('First name is too long.'));
+        }
+
+        if (data.surname.length > 256) {
+            return next(new Error('Last name is too long.'));
+        }
+
+        if (data.username.length > 256) {
+            return next(new Error('Username is too long.'));
+        }
+
+        if (data.email.length > 256) {
+            return next(new Error('Email is too long.'));
+        }
+
+        // Require passwords to be at least 5 characters.
```

Comment:
your comment in userLogin.tsx is
"        // Require passwords to be at least 10 characters."
that's why we need save values as constant